packages/platform/atoms/booker/BookerPlatformWrapper.tsx

```diff
@@ -107,23 +122,28 @@ export const BookerPlatformWrapper = (props: BookerPlatformWrapperAtomProps) =>
     return getUsernameList(username ?? "").length > 1;
   }, [username]);
 
-  const { isSuccess, isError, isPending, data } = useEventType(username, props.eventSlug, props.isTeamEvent);
+  const { isSuccess, isError, isPending, isFetching, data } = useEventType(
+    username,
+    props.eventSlug,
+    props.isTeamEvent,
+    teamId
+  );
   const {
     isSuccess: isTeamSuccess,
     isError: isTeamError,
     isPending: isTeamPending,
-    data: teamData,
-  } = useTeamEventType(props.teamId, props.eventSlug, props.isTeamEvent);
+    data: teamEventTypeData,
+  } = useTeamEventType(teamId, props.eventSlug, props.isTeamEvent);
 
   const event = useMemo(() => {
-    if (props.isTeamEvent) {
+    if (props.isTeamEvent && !isTeamPending && teamId && teamEventTypeData && teamEventTypeData.length > 0) {
```

Comment:
Why adding the checks `&& !isTeamPending && teamId && teamEventTypeData && teamEventTypeData.length > 0` ?

On line 145 we already check for data transformation and if the team is pending (loading) then we don't enter if statement and return non team response.

Reply:
if the teamId changes, we don't want to render previous data while it's fetching new one